# HG changeset patch
# User khazra
# Date 1363811996 25200
# Node ID 7750666eaf166e523bf2bea826ef6ef1056ea252
# Parent  05e5519ffbccc0a6cce2d7cffcd74778aba1378e
8010213: Some api/javax_net/SocketFactory tests fail in 7u25 nightly build
Summary: Eliminate fall-through while setting socket options on Windows
Reviewed-by: alanb, chegar

--- openjdk/jdk/src/windows/classes/java/net/DualStackPlainSocketImpl.java
+++ openjdk/jdk/src/windows/classes/java/net/DualStackPlainSocketImpl.java
@@ -184,14 +184,15 @@
         int optionValue = 0;
 
         switch(opt) {
-            case TCP_NODELAY :
-            case SO_OOBINLINE :
             case SO_REUSEADDR :
                 if (exclusiveBind) {
                     // SO_REUSEADDR emulated when using exclusive bind
                     isReuseAddress = on;
                     return;
                 }
+                // intentional fallthrough
+            case TCP_NODELAY :
+            case SO_OOBINLINE :
             case SO_KEEPALIVE :
                 optionValue = on ? 1 : 0;
                 break;
